src/Suggestor/Suggestor.ts

```diff
@@ -496,14 +497,37 @@ export function onlySuggestIfBracketOpen(fn: SuggestionBuilder, brackets: [strin
  * and false value otherwise.
  *
  * This checks for simple pre-conditions:
+ *  - Does the parent editor explicitly request it?
  *  - Is the global filter (if set) in the line?
- *  - Is the line a task line (with a checkbox)
+ *  - Is the line a task line (with a checkbox)?
  * @param line
- * @param cursorPosition - the cursor position, when 0 is presumed to mean 'at the start of the line'.
- *                          See 'ch' in https://docs.obsidian.md/Reference/TypeScript+API/EditorPosition
+ * @param cursor - the cursor position, when ch is 0 it is presumed to mean 'at the start of the line'.
+ *                          See https://docs.obsidian.md/Reference/TypeScript+API/EditorPosition
+ * @param editor - the editor instance to which the suggest belongs
  */
-export function canSuggestForLine(line: string, cursorPosition: number) {
-    return GlobalFilter.getInstance().includedIn(line) && cursorIsInTaskLineDescription(line, cursorPosition);
+export function canSuggestForLine(line: string, cursor: EditorPosition, editor: Editor) {
+    const lineHasGlobalFilter = GlobalFilter.getInstance().includedIn(line);
+    const didEditorRequest = editorIsRequestingSuggest(editor, cursor, lineHasGlobalFilter);
+
+    if (typeof didEditorRequest === 'boolean') return didEditorRequest;
+    return lineHasGlobalFilter && cursorIsInTaskLineDescription(line, cursor.ch);
+}
+
+/**
+ * Return
+ * - true if the parent editor is explicitly requesting that the suggest be displayed
+ * - false if it is requesting that it be hidden
+ * - undefined if the parent editor wants to defer to the default behavior
+ *
+ * @param editor - the editor instance to which the suggest belongs
+ * @param cursor - the cursor position
+ */
+function editorIsRequestingSuggest(
+    editor: Editor,
+    cursor: EditorPosition,
+    lineHasGlobalFilter: boolean,
+): boolean | undefined {
+    return (editor as any)?.editorComponent?.showTasksPluginAutoSuggest?.(cursor, editor, lineHasGlobalFilter);
```

Comment:
@mgmeyers just a comment here for my knowledge. I'm curious why the `editorComponent` is needed? In Kanban plugin code (https://github.com/mgmeyers/obsidian-kanban/blob/5fa792b9c2157390fe493f0feed6f0bc9be72910/src/components/Editor/MarkdownEditor.tsx#L100-L106) I don't see that being used, what am I missing here? Why not calling just `(editor as any)?.showTasksPluginAutoSuggest?.(cursor, editor, lineHasGlobalFilter);`?